proposals/pr002-hunkspace.md

```diff
@@ -0,0 +1,545 @@
+same mechanism to request other types of localized data that is
+not a part of CLDR.  Such as, for example, a translation of the
+string "Firefox" into Urdu.  While everyone could roll their own
+approach to localization (as was the case historically) , having a
+common specification could go a long way towards a common solution that
+works well in many use cases.  Similarly to how the industry has mostly
+used the ICU library for handling internationalization matters.
+
+# Delegated localization
+
+The language pack approach works reasonably well to support a known set
+of localizable resources.  There are use cases, however, for which a
+program does not know the closed set of localizable resources.
+
+This use case arose recently, again, in Fuchsia OS, related to the work on
+Accessibility Screen Reader. Fuchsia encourages a modular approach to building
+programs, which means that what we consider "apps" internally consists of a
+number of communicating services.  The discussion of
+[FIDL](https://fuchsia.dev/fuchsia-src/development/languages/fidl), the IPC
+method used, is out of scope of this document.
+
+The general pattern that arises is shown in the diagram below, showing
+the collaboration diagram of an actual interaction a user has when using
+audio output from the Accessibility library.
+
+[![Diagram showing interaction with users](pr002/image1.png)]
+
+The Voice UI delegates the choice of "Message!" to a service it does
+not know about in advance, but has to render the message. In the case of
+the voice UI, "rendering of a message" is turning the text to synthetic
+speech; but one can see how a similar situation can arise in other
+aspects of the program lifecycle.  For example, a spell checker service
+needs to report an error to the user.
+
+Let's call the pattern "delegated localization": the localization
+is produced by a component which is different from the component that
+renders the localized resource to the user. The renderer of the
+message does not have access to the to-be rendered message ahead of
+time.
+
+Another instance of delegated localization is an application using
+platform-level localizations for common user interface elements ("OK",
+"Cancel", "Edit"\...).   Mechanisms to do this are reinvented in each
+independent piece of software.  A localization clearinghouse would allow
+the OS to centralize localization decisions.
+
+A way to resolve is to devise some form of a mini "rendering protocol"
+where the screen reader provides rendering hints, and the voice UI
+renderer follows the hints.  The claim here is that this specific
+pattern is common enough that localization could benefit from a common
+specification of that rendering protocol.
+
+# Detailed Proposal
+
+Allow independent third parties to put new non-overlapping points into
+a common hunkspace, such that for each platform a single data provider
+can be used to provide any localized resource.
```

Comment:
Do you have in mind some sort of verification/legitimizing method in mind?

Reply:
While I haven't thought that far just yet, I can see how a DNS-based resolution could work. It could mimic the existing DNS/TLS approaches.  I.e. if you own a domain, you own the namespace.